test/python/algorithms/test_grover.py

```diff
@@ -327,12 +327,14 @@ def test_circuit_result(self, use_sampler):
         result = grover.amplify(problem)
         if use_sampler:
             for i, dist in enumerate(result.circuit_results):
+                print(i, dist)
```

Comment:
Make sure we don't end up with the print here,

Reply:
Thx! Strange values were returned in the CI. I was looking into it, but it seems it's fixed now.

Reply:
Oh, thank you so much. I had no idea and I was stuck!

Reply:
So this is all good here now? The sampler re-use fix was merged and this seems to be passing. Tests you commented out are back again - it all seems fine right?

Reply:
Yes, the test can pass the CI now. Could you please merge this? (I was waiting for the CI, but it took so long)